src/plugins/profiling/server/routes/flamechart.ts

```diff
@@ -69,57 +69,30 @@ async function queryFlameGraph(
       return await client.search(
         {
           index: eventsIndex.name,
-          size: 0,
+          track_total_hits: false,
+          size: 150000, // should be max 100k, but might be more. Better be on the safe side.
           query: filter,
-          aggs: {
-            group_by: {
-              composite: {
-                size: 100000, // This is the upper limit of entries per event index.
-                sources: [
-                  {
-                    traceid: {
-                      terms: {
-                        field: 'StackTraceID',
-                      },
-                    },
-                  },
-                ],
-              },
-              aggs: {
-                count: {
-                  sum: {
-                    field: 'Count',
-                  },
-                },
-              },
-            },
-            total_count: {
-              sum: {
-                field: 'Count',
-              },
-            },
-          },
         },
         {
           // Adrien and Dario found out this is a work-around for some bug in 8.1.
           // It reduces the query time by avoiding unneeded searches.
           querystring: {
             pre_filter_shard_size: 1,
-            filter_path:
-              'aggregations.group_by.buckets.key,aggregations.group_by.buckets.count,aggregations.total_count,_shards.failures',
+            filter_path: 'hits.hits._source.StackTraceID,hits.hits._source.Count,_shards.failures',
           },
         }
       );
     }
   );
 
-  let totalCount: number = resEvents.body.aggregations?.total_count.value;
-  let stackTraceEvents = new Map<StackTraceID, number>();
+  let totalCount: number = 0;
+  const stackTraceEvents = new Map<StackTraceID, number>();
 
   await logExecutionLatency(logger, 'processing events data', async () => {
-    resEvents.body.aggregations?.group_by.buckets.forEach((item: any) => {
-      const traceid: StackTraceID = item.key.traceid;
-      stackTraceEvents.set(traceid, item.count.value);
+    resEvents.hits.hits.forEach((item: any) => {
```

Comment:
When viewing either flamegraph, I received a 500 error that said that `resEvents.hits` does not exist.

Reply:
Hm, I get an error when I try with `resEvents.body.hits...`.
I have a ES 8.3 running... this is not good, we need code that works with both versions.
Will test this with 8.1.

Reply:
We now have some compatibility functions.